assets/blueprints/genesis_helper/src/lib.rs

```diff
@@ -4,178 +4,204 @@ use scrypto::blueprints::epoch_manager::*;
 use scrypto::prelude::*;
 
 // Important: the types defined here must match those in bootstrap.rs
-type AccountIdx = usize;
-type ResourceIdx = usize;
-type ValidatorIdx = usize;
-
-// This data represents data from Olympia Network state
 #[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
-pub struct GenesisData {
-    validators: Vec<GenesisValidator>,
-    resources: Vec<GenesisResource>,
-    accounts: Vec<ComponentAddress>,
-    resource_balances: BTreeMap<ResourceIdx, Vec<(AccountIdx, Decimal)>>,
-    xrd_balances: BTreeMap<AccountIdx, Decimal>,
-    stakes: BTreeMap<ValidatorIdx, Vec<(AccountIdx, Decimal)>>,
+pub struct GenesisValidator {
+    pub key: EcdsaSecp256k1PublicKey,
+    pub accept_delegated_stake: bool,
+    pub is_registered: bool,
+    pub metadata: Vec<(String, String)>,
+    pub owner: ComponentAddress,
 }
 
 #[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
-pub struct GenesisValidator {
-    pub key: EcdsaSecp256k1PublicKey,
-    pub component_address: ComponentAddress,
+pub struct GenesisStakeAllocation {
+    pub account_index: u32,
+    pub xrd_amount: Decimal,
 }
 
 #[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
 pub struct GenesisResource {
-    pub symbol: String,
-    pub name: String,
-    pub description: String,
-    pub url: String,
-    pub icon_url: String,
-    pub address_bytes: [u8; 27],
-    pub owner_with_mint_and_burn_rights: Option<AccountIdx>,
+    pub address_seed_bytes: [u8; 26],
+    pub initial_supply: Decimal,
+    pub metadata: Vec<(String, String)>,
+    pub owner: Option<ComponentAddress>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+pub struct GenesisResourceAllocation {
+    pub account_index: u32,
+    pub amount: Decimal,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+pub enum GenesisDataChunk {
```

Comment:
Would it make better sense if all accounts were stored in a single index rather than per chunk?

Reply:
hmh, but then we'd need a single transaction that lists all accounts? which may be quite sizeable

Reply:
ah, but we could split the index into multiple txns too, but then we'd need to store it in the genesis helper component (in a kv map?) which may ruin the whole benefit

Reply:
I did a quick size test with the current format:
a single genesis txns that was previously ~28.5 MiB, when turned into ~600 txns (granularity tbd) totalled ~37 MiB. Looks quite reasonable to me

Reply:
Ah got it, so each GenesisDataChunk is supposed to correspond to another txn. Makes sense!

Reply:
> so each GenesisDataChunk is supposed to correspond to another txn
yup